File: java/Hardware.java

// deklarasi kelas Hardware dengan inheritance terhadap kelas Product
public class Hardware extends Product{
    
    // deklarasi dan inisialisasi atribut private
    private String brand = "-";
    private String model = "-";

    
    // constructor tanpa parameter
    public Hardware(){
        
    }

    // constructor dengan parameter
    public Hardware(String brand, String model){
        this.brand = brand;
        this.model = model;
    }

    // deklarasi setter sebagai method (Write)
    public void setBrand(String brand){
        this.brand = brand;
    }
    public void setModel(String model){
        this.model = model;
    }

    // deklarasi getter sebagai method (Read Only)
    public String getBrand(){
        return this.brand;
    }
    public String getModel(){
        return this.model;
    }
    
    // deklarasi method Display untuk menampilkan data menggunakan method Getter
    public void displayHardware(){
        System.out.println("Brand        : " + this.getBrand());
        System.out.println("Model        : " + this.getModel());
    }
}

File: c++/Hardware.cpp
#include <iostream>
#include <string>

using namespace std;

// import file kelas parent
#include "Product.cpp"

// deklarasi kelas Hardware dengan inheritance terhadap kelas Product
class Hardware : public Product{
    
    // deklarasi dan inisialisasi atribut private
    private:
        string brand = "-";
        string model = "-";

    public:

        // constructor tanpa parameter
        Hardware(){
            
        }

        // constructor dengan parameter
        Hardware(string inputBrand, string inputModel){
            this->brand = inputBrand;
            this->model = inputModel;
        }

        // deklarasi setter sebagai method (Write)
        void setBrand(string brand){
            this->brand = brand;
        }
        void setModel(string model){
            this->model = model;
        }

        // deklarasi getter sebagai method (Read Only)
        string getBrand(){
            return this->brand;
        }
        string getModel(){
            return this->model;
        }

        // deklarasi method Display untuk menampilkan data menggunakan method Getter
        void displayHardware(){
            
            cout << "Brand        : " << this->getBrand() << endl; 
            cout << "Model        : " << this->getModel() << endl; 

        }

        // destructor
        ~Hardware(){

        }

};